src/main/scala/zio/prelude/NonEmptySet.scala

```diff
@@ -0,0 +1,207 @@
+package zio.prelude
+
+import zio.NonEmptyChunk
+
+import scala.language.implicitConversions
+
+final class NonEmptySet[A] private (val toSet: Set[A]) { self =>
+
+  /**
+   * Returns an element of this `NonEmptySet` and the remainder, which is a (possibly empty) `Set`.
+   */
+  def destruct: (A, Set[A]) = (toSet.head, toSet.tail)
+
+  /**
+   * Converts this `NonEmptySet` to a `NonEmptyChunk`.
+   */
+  def toNonEmptyChunk: NonEmptyChunk[A] = destruct match { case (head, tail) => NonEmptyChunk.fromIterable(head, tail) }
+
+  /**
+   * Converts this `NonEmptySet` to a `NonEmptyList`.
+   */
+  def toNonEmptyList: NonEmptyList[A] = destruct match { case (head, tail) => NonEmptyList.fromIterable(head, tail) }
+
+  /** Creates a new `NonEmptySet` with an additional element, unless the element is
+   *  already present.
+   *
+   *  @param elem the element to be added
+   *  @return a new set that contains all elements of this set and that also
+   *          contains `elem`.
+   */
+  def +(elem: A): NonEmptySet[A] = new NonEmptySet(toSet + elem)
+
+  /** Creates a new `NonEmptySet` with additional elements, omitting duplicates.
+   *
+   *  This method takes two or more elements to be added. Elements that already exist in the `NonEmptySet` will
+   *  not be added. Another overloaded variant of this method handles the case where a single element is added.
+   *
+   *  Example:
+   *   {{{
+   *    scala> val a = NonEmptySet(1, 3) + 2 + 3
+   *    a: zio.prelude.NonEmptySet[Int] = NonEmptySet(1, 3, 2)
+   *   }}}
+   *
+   *  @param elem1 the first element to add.
+   *  @param elem2 the second element to add.
+   *  @param elems the remaining elements to add.
+   *  @return   a new `NonEmptySet` with the given elements added, omitting duplicates.
+   */
+  def +(elem1: A, elem2: A, elems: A*): NonEmptySet[A] = new NonEmptySet(toSet + elem1 + elem2 ++ elems)
```

Comment:
I think this `+` for adding multiple elements syntax has been removed in the Scala 2.13 collections rework so we probably should drop as well since we already have `+` and `++`.